### PR TITLE
Setting up SmileID and Networking Interfaces 👷🏽 

### PR DESCRIPTION
**Pull request by jumaallan**
### **User description**
Story: https://app.shortcut.com/smileid/story/xxx

## Summary

A few sentences/bullet points about the changes

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

___

### **PR Type**
Enhancement

___

### **Description**
- Implemented SmileID KMP interface

- Created model classes for API requests/responses

- Added mappers for Android platform implementation

- Replaced SmileIdentity with SmileID implementation

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>23 files</summary><table>
<tr>
  <td><strong>PlatformDependency.kt</strong><dd><code>Add Android platform dependency class</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-85913f7506547909af8a5b054d0304ff6b0211a48e80ae9a042bcad214554964">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileID.kt</strong><dd><code>Implement Android-specific SmileID functionality</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-99772498552e655b3283c05ab63bbb4af0d29c72aff849ec39ddd6d37aac2a7c">+92/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIdentity.kt</strong><dd><code>Remove old SmileIdentity implementation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-ec3bce0247563b60fe81b7add6c8c70c0919d3941e7c040fa0de13a24ea5c0cf">+0/-12</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Mapper.kt</strong><dd><code>Create mappers for Android models</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-670ac56e8c7dbdf6f64a68de3fd99d62437aa21d7e2ce4cb7464a8a5a104629a">+123/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>PlatformDependency.kt</strong><dd><code>Define common platform dependency interface</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-eb240c205f1c065ae2602079eba3605ca1f5d6335e8962015c325ffa8be5a5d4">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileID.kt</strong><dd><code>Define common SmileID interface</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-c90049aecbc61d642ab8663c9533c9a91cd2ea0b516ee1c92d8f204170e03404">+39/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIdentity.kt</strong><dd><code>Remove old common SmileIdentity interface</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-eb3022afa887593937d0b0fa159d4ff58aac326607c782f2a33d49c26e2f1968">+0/-5</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Authentication.kt</strong><dd><code>Add authentication models</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-df717b6456b75170eefa39838a17e7d4009811d5af998e5c6fe350adf000cddf">+39/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>KmpConfig.kt</strong><dd><code>Add configuration model</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-516b23f59eb34c86f7a4730dccb57269bd8275da45048be96428e21f51b69c28">+12/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>KmpConsentInformation.kt</strong><dd><code>Add consent information model</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-f0c304338d3a35681315473287d476960d73296fd60a6d1fc4504db73b98f47d">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>KmpEnhancedKyc.kt</strong><dd><code>Add enhanced KYC models</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-a112fd3483f7fd0c0518df44717db6f0024bfc6893e5daeba2671128a818c20e">+26/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>PartnerParams.kt</strong><dd><code>Add partner parameters and job type models</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-d5b96f9be6f29444a0d6ee8794456afba3f5037dfba68154c0862c7c9a85a122">+33/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>PlatformDependency.kt</strong><dd><code>Add iOS platform dependency implementation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-29fb934eddcc289e3a30e51e4013949d7338dff91bbd36c2c7884ddfe8932987">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileID.kt</strong><dd><code>Add iOS SmileID implementation stubs</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-e56c98873ae4ff6b7800608ba62497c7429e90a2380e297479fa4e7b06b7deee">+44/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIdentity.kt</strong><dd><code>Remove old iOS SmileIdentity implementation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-2bbc3f4fad69067e27764123374d78e9884b1e5b660cf85c0029c569a34bc027">+0/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>PlatformDependency.kt</strong><dd><code>Add JVM platform dependency implementation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-6ddd2f6a4769d34042dcfe2500ba17dbc950022f63cc0c8e15e87161d0b9afc0">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileID.kt</strong><dd><code>Add JVM SmileID implementation stubs</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-de7e18fc2779dec1a4f382ce1062d9123b14f014415e07b2ab7264c0ce4d1e9c">+44/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIdentity.kt</strong><dd><code>Remove old JVM SmileIdentity implementation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-988fbb48a6051e20a1b6840d8fd868b204ddf29dabdeb2bf4e6a0aca9d20b939">+0/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>PlatformDependency.kt</strong><dd><code>Add Linux platform dependency implementation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-d17552caf4ae38948deea5b884f210c5c8980f5a4b72d5a9d353884e7b66e1d1">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileID.kt</strong><dd><code>Add Linux SmileID implementation stubs</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-362a2e1d57f0c40c115ee07e2e253b800509850ae4e11ae2c4e251170b99d2e6">+44/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIdentity.kt</strong><dd><code>Remove old Linux SmileIdentity implementation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-ec8670024c5d8b3069b5babbf26ed6240593c7bd4870f7ba83c6e05e5c92645a">+0/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>MainActivity.kt</strong><dd><code>Add sample SmileID API usage</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-e91e2320e6b44d0eeac2c954b5f7b4e4d5a8e0c348dd82d857fd1ba4bb570a04">+44/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIDApplication.kt</strong><dd><code>Update application to use new SmileID API</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-63da44338dfdf0770c5082a53a455bdaebe6a46c73ea5349abfc4b6d52a76c5c">+6/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Additional files</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>App.kt</strong></td>
  <td><a href="https://github.com/smileidentity/kmp/pull/10/files#diff-b211d8387b9d61d17ac1f69bcaa14e78864d98508cb00056fffcc862be57ac97">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

File: sample/cmp/app/src/androidMain/kotlin/com/smileidentity/kmp/sample/MainActivity.kt [hunk 1]
```diff
@@ -4,13 +4,57 @@ import android.os.Bundle
...
+import com.smileidentity.kmp.models.KmpPartnerParams
+import java.util.UUID
 
 class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         setContent {
+
+            // todo - will delete this
```
Comment (MamboBryan) [inline sample/cmp/app/src/androidMain/kotlin/com/smileidentity/kmp/sample/MainActivity.kt:25]: 😆  thanks for clarifying this @jumaallan. Looks great to me 🥳
